### ISSUE TITLE
False positive scan test for the OVAL feed

### ISSUE THREAD

**Issue by jmv74211** [wazuh/wazuh-qa#1022]
Hello team,

Doing several checks in the test scan for the vulnerability detector module, I noticed the tests fail because the following log is not found:
```
The OVAL found a total of '5' potential vulnerabilities for agent '000'
```

See the following error case:

![image](https://user-images.githubusercontent.com/23462183/106469180-40857d80-649f-11eb-8b67-08f4bb96a386.png)

On the left, we can see that the NVD scan log appears but not the OVAL. For example, and expected log case would be:

![image](https://user-images.githubusercontent.com/23462183/106469337-71fe4900-649f-11eb-8a76-139d7b4696d6.png)

This ocurrs randomly, so there is no way to reproduce it, just run it several times.

I have encountered this error case in:

- `test_ubuntu_inventory_canonical_feed.py`
- `test_redhat_inventory_redhat_feed.py` 
-  `test_debian_inventory_debian_feed.py`

One possible reason for this error, is that the log is truncated just after the log is recorded.

It is requested to investigate this case and solve it.

Best regards.

**Comment by jmv74211** [comment]
I have been investigating why there are times when the tests pass and times when they don't, apparently randomly.

First of all, I have been monitoring the `ossec.log` with `modulesd.DEBUG=2` (which is what the test checks) to see if when that failure occurs in the test, the events are recorded in the `ossec.log` or not.

After repeating the process several times until I get a failure, I noticed that the events do not appear in the log, so the test fails.

For example, it looks for events like the following:

```
DEBUG: (5468): The 'vulnerable_package' package (1.0) from agent '000' is vulnerable to 'CVE-024'. Condition: 'Package less than 1.5.0'
``` 

Now I am wondering the following. *Why are there times when events are recorded in the log and times when they are not?*

I have been discussing this with @BraulioV , and he has asked this to the core team. The core team has confirmed that it is possible that there are race conditions when writing in the log  (especially when there is `DEBUG = 2`). This has given us to think that there may be times that messages are not being written in the log even though the alerts are being generated.

What we have done next has been to eliminate all the ruleset rules not corresponding to vuln detector, truncate the `alerts.json` and run tests until they fail to compare the number of alerts that should have been generated and those registered in the `alerts.json`.

Well, it seems that when the test fails, neither the corresponding logs are generated, nor the alerts are generated, so for now **we discard the possibility that the event does not appear in the log but the alerts are generated**.

In the meantime, we continue to investigate this further.

**Comment by jmv74211** [comment]
I have been investigated more about this, and for now, I have not found the possible cause for these false-positive alerts since I have not reached to reproduce the test failures. 

I have been discussing with the QA team about the vuln scan tests structure, and we have been talking about changing the way of mocking the data (there may be times when data is not imported correctly and the test fails). 

For this reason, I have opened this issue https://github.com/wazuh/wazuh-qa/issues/1050 that will be reviewed later.

I will keep an eye out in case I see a failed test again.

**Comment by jmv74211** [comment]
Hi,

During this time, the QA team has been solving other issues related to vuln detector. In parallel, I have been investigating the cause of these errors and it appears that these errors could be caused by the following causes:

- Sometimes, the `wazuh-modulesd` proccess was not killed successfully,so it was duplicated causing that the test fails. This has been solved in https://github.com/wazuh/wazuh-qa/pull/1058.

- Until now, we used the agent ' 000'  for scanning testing. This caused some code dependencies such us can not stop `wazuh-modulesd` before mocking the agent system. Now all mocking process is done after stopping the `wazuh-db` and `wazuh-modulesd` processes. After this, we have refactored the code of system mocking. This has been done in https://github.com/wazuh/wazuh-qa/pull/1058

After applying these changes, we have not observed more false-positive errors.

We can close this issue for now, and re-open it in the future if necessary.

Regards.

**Comment by jmv74211** [comment]
Hi,

During this time, the QA team has been solving other issues related to vuln detector. In parallel, I have been investigating the cause of these errors and it appears that these errors could be caused by the following causes:

- Sometimes, the `wazuh-modulesd` proccess was not killed successfully,so it was duplicated causing that the test fails. This has been solved in https://github.com/wazuh/wazuh-qa/pull/1058.

- Until now, we used the agent ' 000'  for scanning testing. This caused some code dependencies such us can not stop `wazuh-modulesd` before mocking the agent system. Now all mocking process is done after stopping the `wazuh-db` and `wazuh-modulesd` processes. After this, we have refactored the code of system mocking. This has been done in https://github.com/wazuh/wazuh-qa/pull/1058

After applying these changes, we have not observed more false-positive errors.

We can close this issue for now, and re-open it in the future if necessary.

Regards.
